### PR TITLE
fix(tlog): fail closed for rekor v2 parsing

### PR DESCRIPTION
**Pull request by 1seal**
summary:
- tighten rekor v2 parsing to require `apiVersion: "0.0.2"` and a supported v2 `spec` oneof; otherwise fall back to rekor v1 parsing
- make `ValidateEntry()` fail closed when a rekor v2 entry has an unset/unsupported `spec` oneof
- add tests covering v1 fallback and v2 empty-spec rejection

### CODE REVIEW

File: pkg/tlog/entry.go [hunk 1]
```diff
@@ -525,9 +523,16 @@ func unmarshalRekorV1Entry(body []byte) (types.EntryImpl, error) {
 
 func unmarshalRekorV2Entry(body []byte) (*rekortilespb.Entry, error) {
 	logEntryBody := rekortilespb.Entry{}
-	err := protojson.Unmarshal(body, &logEntryBody)
-	if err != nil {
+	if err := protojson.Unmarshal(body, &logEntryBody); err != nil {
+		return nil, ErrInvalidRekorV2Entry
+	}
+	if logEntryBody.GetApiVersion() != "0.0.2" {
```
Comment (kommendorkapten) [inline pkg/tlog/entry.go]: I think we should switch the logic around here, first get the entry type, then check the api version based on that as the entry type is the factor that decides which api versions that are allowed.

**Comment by 1seal** [comment]
agreed — i updated unmarshalRekorV2Entry to first determine the v2 spec type (oneof) and then validate apiVersion against the allowed version(s) for that type, since entry type is what drives the version constraint.

added a regression test for a valid v2 spec with a mismatched apiVersion (now rejected). pushed in d57cc40.

**Comment by kommendorkapten** [comment]
Thank you! The change looks good, can you just ammend the commit with signoff?

**Comment by 1seal** [comment]
done — amended with dco signoff and force-pushed. new commit: 8bd066d.

**Review by kommendorkapten — DISMISSED** [review]
Thank you, looks good.

Will wait for @Hayden-IO to give a look as well, as this is code he is more familiar with than I am.

**Review by Hayden-IO — DISMISSED** [review]
Thanks, this LGTM! Can you rebase?

**Comment by 1seal** [comment]
thanks, updated. the cap is now enforced in (*bundle.Bundle).TlogEntries() (before make(...)/parsing), with the same too many verificationMaterial.tlogEntries: got=%d max=%d error format

**Comment by 1seal** [comment]
done — rebased onto latest origin/main and force-pushed. new commits: 7036375, a45b041.

**Review by Hayden-IO — APPROVED** [review]
Thanks!
